Islandora-CLAW/controlled_access_terms
# content types become taxonomies

**GitHub Issue**: 

- [#902](https://github.com/Islandora-CLAW/CLAW/issues/902): Convert controlled_access_terms config into a feature
- [#912](https://github.com/Islandora-CLAW/CLAW/issues/912): controlled_access_terms content types as taxonomies
- [#915](https://github.com/Islandora-CLAW/CLAW/issues/915): Fix controlled_access_terms TypedRelations validateValues

# What does this Pull Request do?

This PR incorporates @whikloj's work moving the configs into a feature, a bug-fix for the TypedRelations field, and turning the Agents and Subjects into taxonomy vocabularies. 

# How should this be tested?

From a clean install:

- Apply the PR
- Enabling the main Controlled Access Terms module will make the TypedRelation and AuthorityLink available.
- Enabling the Default Configuration sub-module will install the configs to create taxonomy vocabularies for Corporate Bodies, Persons, Families, Geographic Locations, and Subjects.
- Creating a term for any of them should not cause errors.
- Also, adding a TypedRelation field and then updating the default available relation types shouldn't WSOD. 
 
# Additional Notes:

I neglected to add a typed relation field before issuing the PR. One will come shortly.

This PR supersedes #6 and #7.

# Interested parties
@Islandora-CLAW/committers (esp. @whikloj and @Natkeeran)
